docs/content/guides/dagster/using-environment-variables-and-secrets.mdx

```diff
@@ -0,0 +1,199 @@
+---
+title: Using environment variables and secrets | Dagster Docs
+description: Learn how to securely define and access environment variables in your Dagster code.
+---
+
+# Using environment variables and secrets
+
+<EnvironmentVariablesIntro />
+
+---
+
+## Declaring environment variables
+
+How environment variables are declared depends on how your Dagster project is deployed.
+
+### Dagster Cloud
+
+Environment variables can be set a variety of ways in Dagster Cloud, including directly in the UI. Refer to the [Dagster Cloud environment variables guide](/dagster-cloud/developing-testing/environment-variables-and-secrets) for more info.
+
+### Dagster Open Source
+
+Setting environment variables in Dagster Open Source depends on where Dagster is deployed. Refer to the deployment guide for your platform for more info:
+
+<ArticleList>
+  <ArticleListItem
+    title="Amazon Web Services EC2 / ECS"
+    href="/deployment/guides/aws#secrets-management-in-ecs"
+  ></ArticleListItem>
+  <ArticleListItem
+    title="Docker"
+    href="/deployment/guides/docker"
+  ></ArticleListItem>
+  <ArticleListItem
+    title="Kubernetes"
+    href="/deployment/guides/kubernetes/customizing-your-deployment#security"
+  ></ArticleListItem>
+</ArticleList>
+
+---
+
+## Syntax
+
+To access a configured environment variable in your Dagster code, use the following syntax:
+
+```python
+"PARAMETER_NAME": {"env": "ENVIRONMENT_VARIABLE_NAME"}
```

Comment:
I don't think this is how you access it in Python code exactly. You access an environment variable in Python code like this:

```
import os
my_env_var = os.getenv("ENVIRONMENT_VARIABLE_NAME")
```

What you have here is how you bring environment variables into Dagster config using the StringSource type. For example, If i defined an op like 

```
@op(config_schema={"param_name": StringSource})
def my_op(context):
  name = context.op_config["param_name"]
```

I could then configure it in the launchpad like this:
```
ops:
  my_op:
    config:
       param_name:
         env:
           ENVIRONMENT_VARIABLE_NAME  
```

We should separate out how to access the environment variables in Python code (which isn't particularly Dagster-specific) with how to incorporate them into the Dagster type system (which is optional and very Dagster-specific)